pkg/proxy/service.go

```diff
@@ -129,6 +141,10 @@ func (sct *ServiceChangeTracker) newBaseServiceInfo(port *v1.ServicePort, servic
 	if apiservice.RequestsOnlyLocalTraffic(service) {
 		onlyNodeLocalEndpoints = true
 	}
+	onlyNodeLocalEndpointsForInternal := false
+	if apiservice.RequestsOnlyLocalTrafficForInternal(service) {
```

Comment:
this needs to check the gate - it's possible the gate may be disabled but the service has old data in it, we still don't want to use that  data (but do preserve it)

Reply:
added gate check logic before `apiservice.RequestsOnlyLocalTrafficForInternal(service)`